include/networkit/distance/DynDijkstra.hpp

```diff
@@ -39,8 +41,21 @@ class DynDijkstra final : public DynSSSP {
     void updateBatch(const std::vector<GraphEvent> &batch) override;
 
 private:
-    enum Color { WHITE, BLACK };
+    enum Color { WHITE, GRAY, BLACK };
     std::vector<Color> color;
+    const edgeweight infDist = std::numeric_limits<edgeweight>::max();
+
+    struct Compare {
+    public:
+        Compare(const std::vector<double> &dist_) : dist(dist_) {}
+
+        bool operator()(node x, node y) const { return dist[x] < dist[y]; }
+
+    private:
+        const std::vector<double> &dist;
+    };
+
+    tlx::d_ary_addressable_int_heap<node, 2, Compare> heap;
```

Comment:
Here you can use the vector comparators already implemented in `auxiliary/VectorComparator.hpp`